### PR TITLE
docs: Update WCT documentation with latest v7 & v8 details and single project setup

### LINKED ISSUE #16872: [Docs] Windows Community Toolkit documentation will need to be updated to add latest details
Windows Community Toolkit documentation will need to be updated to add latest details:
- [ ] Details about the setup with the new single project structure
- [ ] Details about the migration from v7 to v8 for WCT
- [ ] Details about direct support for Uno Platform, simplifying the setup process starting with WCT v8
___

> Hi @tomchoondal (cc @palusi, @arnold256) 
> We will update soon [our documentation](https://platform.uno/docs/articles/uno-community-toolkit.html?tabs=tabid-winui) with the new changes.

> But here are the details with our [latest single project structure](https://platform.uno/blog/the-first-and-only-true-single-project-for-mobile-web-desktop-and-embedded-in-net/):

- For WCT (**v. 7.1.2 and below**):
	`xmlns:controls="using:CommunityToolkit.WinUI.UI.Controls"`
	
	In Directory.Packages.props:
	```
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
	  <PackageVersion Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
	  <!-- Add more community toolkit references here -->
	</ItemGroup>
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
	  <PackageVersion Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.200" />
	  <!-- Add more uno community toolkit references here -->
	</ItemGroup>
	```
	
	In YourProjectName.csproj:
	```
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
	  <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" />
	  <!-- Add more community toolkit references here -->
	</ItemGroup>
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
	  <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" />
	  <!-- Add more uno community toolkit references here -->
	</ItemGroup>
	```
 
- For WCT (**v. 8.0 and above**):
  `xmlns:controls="using:CommunityToolkit.WinUI.Controls"`
  - **NOTE 1: The WCT 8.0 packages and above now include direct support for Uno Platform, simplifying the setup process.**
  - **NOTE 2: Updating to the latest WCT version requires an update to net8.0-windows10.0.22621 and Microsoft.WindowsAppSDK to the latest version (1.5.240311000).**
    (Related WCT issue: https://github.com/CommunityToolkit/Tooling-Windows-Submodule/issues/185)
	
	(Example using _CommunityToolkit.WinUI.Controls.Primitives_ here as I tested the WCT `WrapPanel`)
	In Directory.Packages.props:
	```
	 <ItemGroup>
	    <PackageVersion Include="CommunityToolkit.WinUI.Controls.Primitives" Version="8.1.240328-rc" />
	    <!-- Add more community toolkit references here -->
	 </ItemGroup>
	```
	
	In YourProjectName.csproj:
	```
	  <ItemGroup>
	    <PackageReference Include="CommunityToolkit.WinUI.Controls.Primitives" />
	    <!-- Add more community toolkit references here -->
	  </ItemGroup>
	```
 
> I have made two sample apps to help the [community on discord](https://discord.com/channels/1182775715242967050/1182775715729526908/1237939787176869968) the other week for reference :
> - Sample app using WCT (**v. 7.1.2**): [TestWrapPanelButtons.zip](https://github.com/unoplatform/uno/files/15425484/TestWrapPanelButtons.zip)
> - Sample app using WCT (**v. 8.1.240328-rc**): [TestWrapPanelButtonsWCTRC.zip](https://github.com/unoplatform/uno/files/15425488/TestWrapPanelButtonsWCTRC.zip)

> [!IMPORTANT]
> **Here is the [Migration Guide from v7 to v8 for WCT](https://github.com/CommunityToolkit/Windows/wiki/Migration-Guide-from-v7-to-v8) for additional information on what changed lately between these versions.**
> As for example:
> - For the `DataGrid` control you will need to use **WCT version 7.1.2** as it is no longer a control available in the latest versions (v. 8.0 and above) of WCT. ([related details on Discord here](https://discord.com/channels/1182775715242967050/1193995186951426048/1221955960159993926))
> - For the `Carousel` control you will need to use **WCT version 7.1.2** as well as it is no longer a control available in the latest versions (v. 8.0 and above) of WCT. ([related details on Discord here](https://discord.com/channels/1182775715242967050/1193995186951426048/1235393057055182919))
>
> Also here are the [v. 8.1.240328-rc release notes](https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240328-rc)

_Originally posted by @agneszitte in https://github.com/unoplatform/uno/discussions/13090#discussioncomment-9541265_

### PR DESCRIPTION
**Pull request by agneszitte**
GitHub Issue (If applicable): closes #16872

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## Description

Windows Community Toolkit documentation updated to add the latest details:

- [X]  Details about the setup with the new single project structure and legacy templates for v7 and v8
- [X]  Details about the migration from v7 to v8 for WCT
- [X]  Details about direct support for Uno Platform, simplifying the setup process starting with WCT v8
- [X]  Details about the needed update to Windows SDK 10.0.22621 and above and Microsoft.WindowsAppSDK to the latest version for WCT version 8.x
- [X]  Details about the [new updated related samples](https://github.com/unoplatform/Uno.Samples/pull/757)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/5.2

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/5.3

File: doc/articles/toc.yml [hunk 1]
```diff
@@ -81,7 +81,12 @@
       - name: How to deploy a WebAssembly app on Azure Static Web Apps
         href: xref:Uno.Tutorials.AzureStaticWepApps
       - name: How to use Windows Community Toolkit
-        href: xref:Uno.Development.CommunityToolkit
+        topicHref: xref:Uno.Development.CommunityToolkit
+        items:
+          - name: Windows Community Toolkit - Version 7.x
+            href: xref:Uno.Development.CommunityToolkit.v7
+          - name: Windows Community Toolkit - Version 8.x
+            href: xref:Uno.Development.CommunityToolkit.v8
```
Comment (jeromelaban) [inline doc/articles/toc.yml:89]: Let's invert, latest version being always first.

File: doc/articles/uno-community-toolkit.md [hunk 2]
```diff
@@ -4,206 +4,19 @@ uid: Uno.Development.CommunityToolkit
...
-Controls\TextBox\Themes\Generic.xaml : Xaml Internal Error error WMC9999: 
-Type universe cannot resolve assembly: Uno.UI, Version=255.255.255.255, 
-Culture=neutral, PublicKeyToken=null.
-```
-
-This means that there's an unconditional reference to Uno Platform's packages, and you'll need to make sure to add the conditional references as suggested above.
-
-## Task 1 - Add Windows Community Toolkit to Uno Projects
+- **[For WCT version 7.x](xref:Uno.Development.CommunityToolkit.v7):** The `DataGrid` control, but the same steps can be followed for any of the Uno ported Windows Community Toolkit components.
+- **[For WCT version 8.x](xref:Uno.Development.CommunityToolkit.v8):** The `SettingsCard` control, but the same steps can be followed for any of the Windows Community Toolkit components supported out of the box.
```
Comment (jeromelaban) [inline doc/articles/uno-community-toolkit.md:12]: v8 first

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**